geocity/views.py

```diff
@@ -86,7 +86,55 @@ def get_context_data(self, **kwargs):
         return context
 
 
+def get_redirect_url_after_login(qs_dict):
+    url_value = (
+        qs_dict["next"][0]
+        if "next" in qs_dict
+        else reverse("permits:permit_request_select_administrative_entity")
+    )
+    return url_value
+
+
+def create_duo_client(username, request):
+    # Initial value of duo_client
+    duo_client = None
+
+    # Get user corresponding to the user name
+    user = User.objects.get(username=username)
+
+    # Get base queryset
+    permitdepartement_qs = models.PermitDepartment.objects.filter(
+        group__in=user.groups.all(), mandatory_2fa=True, duo_config__isnull=False
+    )
```

Comment:
Move this query to PermitDepartment Model as a method for a DRYer approach ?
From @monodo on another branch

Reply:
Yes

Reply:
Done